cmd/registry/patch/artifact_manifest.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package patch
+
+import (
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/registry/names"
+	"google.golang.org/protobuf/proto"
+)
+
+const ManifestMimeType = "application/octet-stream;type=google.cloud.apigeeregistry.v1.controller.Manifest"
+
+type ManifestDependency struct {
+	Pattern string `yaml:"pattern"`
+	Filter  string `yaml:"filter,omitempty"`
+}
+
+type ManifestGeneratedResource struct {
+	Pattern      string                `yaml:"pattern"`
+	Filter       string                `yaml:"filter,omitempty"`
+	Receipt      bool                  `yaml:"receipt,omitempty"`
+	Dependencies []*ManifestDependency `yaml:"dependencies"`
+	Action       string                `yaml:"action"`
+}
+
+type Manifest struct {
+	Header `yaml:",inline"`
+	Data   struct {
+		DisplayName        string                       `yaml:"displayName,omitempty"`
+		Description        string                       `yaml:"description,omitempty"`
+		GeneratedResources []*ManifestGeneratedResource `yaml:"generatedResources"`
+	} `yaml:"data"`
+}
```

Comment:
These don't seem like they should be exported structs/consts. They should have documentation otherwise.

Reply:
At this stage, I'm not ready to make them private - I think application code might use them to construct these YAML files. I'd like to hold on documentation until we have more experience with this.

Reply:
I feel like we have opposite defaults for visibility. IMO we should default to private and only export things when they're actually being used because increasing visibility is always "free" - while you can only decrease visibility if you know it isn't used at a higher level of visibility.